app-backend/batch/src/main/scala/com/azavea/rf/batch/airflow/export/CreateExportDef.scala

```diff
@@ -0,0 +1,178 @@
+package com.azavea.rf.batch.airflow.export
+
+import java.util.UUID
+
+import scala.collection.JavaConverters._
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.{Await, Future}
+import scala.concurrent.duration._
+import scala.util._
+
+import akka.actor.ActorSystem
+import akka.stream.ActorMaterializer
+import cats.data._
+import cats.implicits._
+import com.amazonaws.auth.DefaultAWSCredentialsProviderChain
+import com.amazonaws.services.elasticmapreduce.{AmazonElasticMapReduceClient}
+import com.amazonaws.services.elasticmapreduce.model.{AddJobFlowStepsRequest, StepConfig, HadoopJarStepConfig}
+import com.amazonaws.services.route53._
+import com.amazonaws.services.route53.model._
+import io.circe.generic.JsonCodec
+import io.circe.syntax._
+
+import com.azavea.rf.batch.JobNoEC
+import com.azavea.rf.batch.util._
+import com.azavea.rf.database.{Database => DB}
+import com.azavea.rf.database.tables._
+import com.azavea.rf.datamodel._
+
+case class CreateExportDef(exportId: UUID)(implicit val database: DB) extends JobNoEC {
+  val name = "create_export_def"
+  implicit val system       = ActorSystem(s"$name-system")
+  implicit val materializer = ActorMaterializer()
+    
+  protected def writeExportDefToS3(exportDef: ExportDefinition) = {
+    logger.info(s"Uploading export definition ${exportDef.id.toString} to S3 at ${exportDefConfig.bucketName}")
+    val result = Either.catchNonFatal(
+      S3.putObject(s"rasterfoundry-development-data-us-east-1/${exportDefConfig.bucketName}", s"${exportDef.id.toString}.json", None, exportDef.asJson.toString))
+    result match {
+      case Right(_) => s"rasterfoundry-development-data-us-east-1/${exportDefConfig.bucketName}"
+      case Left(ex) => {
+        logger.info(ex.toString())
+        throw new Exception("...")
+      }
+    }
+  }
+
+  def wrapTry[A,B](fn: A => B): A => Option[B] = {
+    (x) => {
+      val result = Either.catchNonFatal(fn(x))
+      result match {
+        case Right(r) => r.some
+        case Left(ex) => {
+          logger.error(ex.toString())
+          None
+        }
+      }
+    }
+  }
+
+  def fx[A,B](fn: A => B): A => Option[A] = {
+    (x) => {
+      wrapTry((y:A) => fn(y)).apply(x) match {
+       case Some(_) => x.some
+       case None => None
+      }
+    }
+  }
+
+  protected def getClusterId() = {
```

Comment:
Agree here, that would help to avoid some complexity.

Smth that i discovered via google search: https://github.com/gilt/scala-srv-dns (who knows how it works)
Working java lib: http://www.xbill.org/dnsjava/dnsjava-current/examples.html